### PR TITLE
Toast service rework

### PR DESCRIPTION
**Pull request by JulienVannier66**
### What does this PR do?

New toast rework

Related to: https://github.com/upfluence/backlog/issues/1631

### What are the observable changes?
Peek 2022-05-17 15-40
Screenshot from 2022-05-17 13-02-25

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [ ] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: .storybook/preview-head.html [hunk 1]
```diff
@@ -7,13 +7,13 @@
...
               var defaultPort = location.protocol === 'https:' ? 443 : 80;
-              var port = 4200;
+              var port = 7020;
               var path = '';
               var prefixURL = '';
               var src = srcUrl || prefixURL + '/_lr/livereload.js?port=' + port + '&host=' + host + path;
               var script = document.createElement('script');
               script.type = 'text/javascript';
-              script.src = location.protocol + '//' + host + ':4200' + src;
+              script.src = location.protocol + '//' + host + ':7020' + src;
```
Comment (Miexil) [inline .storybook/preview-head.html:16]: meh, this is to conflict 100% of the time on my machine ; how about 4201?

File: addon/services/toast.ts [hunk 2]
```diff
@@ -85,123 +93,179 @@ export default class Toast extends Service {
...
+    const counterDownKeyframes = new KeyframeEffect(element, [{ width: '100%' }, { width: '0' }], {
+      duration: duration,
+      fill: 'forwards'
+    });
 
-    if (title && !isEmpty(title)) {
-      toastChildren.splice(0, 0, this._buildElement('div', ['toast-title'], title));
-    }
+    return new Animation(counterDownKeyframes, document.timeline);
+  }
```
Comment (Miexil) [inline addon/services/toast.ts:104]: Nice 👍

File: addon/services/toast.ts [hunk 3]
```diff
@@ -85,123 +93,179 @@ export default class Toast extends Service {
...
+      } else if (opts.badge?.text) {
+        badgeContent = `<span class="upf-badge__text">${opts.badge.text}</span>`;
+      }
+
+      const badgeContainer: HTMLElement = this._buildElement(
+        'div',
+        ['upf-badge', 'upf-badge--shape-round', 'upf-badge--size-md'],
+        badgeContent
+      );
+      mainContainer.append(badgeContainer);
```
Comment (Miexil) [inline addon/services/toast.ts:153]: Maybe move this logic into another private method for better readability ? 
You might be able to do the same for the code below as well ?

File: addon/services/toast.ts [hunk 4]
```diff
@@ -85,123 +93,179 @@ export default class Toast extends Service {
...
 
-    closeButton.classList.add('toast-close-button');
-    closeButton.innerHTML = '<i class="fas fa-times text-size-5"></i>';
-    closeButton.addEventListener('click', this._onToastClose.bind(this), { once: true });
+    toast.addEventListener('mouseenter', () => {
+      this._toasts.get(toast)?.pause();
+    });
 
-    return closeButton;
+    toast.addEventListener('mouseenter', this._pauseCounterAnimation.bind(this));
```
Comment (Miexil) [inline addon/services/toast.ts:241]: Is my github drunk or are you listening for mouseenter twice ? 🤔  (line 237 & line 241 ?)

File: app/styles/toast.less [hunk 5]
```diff
@@ -1,60 +1,114 @@
...
+    background-color: var(--color-gray-50);
+    box-shadow: var(--box-shadow-lg);
 
-  .toast--visible {
-    animation: slide-up 0.5s 1;
-    z-index: 20;
-    display: flex;
-  }
+    .counter {
+      height: 3px;
```
Comment (Miexil) [inline app/styles/toast.less:33]: `@padding-xx-sm / 2`
:trollface:

File: tests/unit/services/toast-test.ts [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import { module, test } from 'qunit';
 import { setupTest } from 'ember-qunit';
-import { click } from '@ember/test-helpers';
+// import { click } from '@ember/test-helpers';
```
Comment (phndiaye) [inline tests/unit/services/toast-test.ts:3]: remove comment :)

File: addon/services/toast.ts (hunk 4 above)
Comment (JulienVannier66) [reply addon/services/toast.ts:241]: You are not drunk, it is me here 🙃

File: app/styles/toast.less (hunk 5 above)
Comment (JulienVannier66) [reply app/styles/toast.less:33]: For this one, it is `@padding-xxx-sm / 2` xD (missing one "x")

File: tests/unit/services/toast-test.ts (hunk 6 above)
Comment (JulienVannier66) [reply tests/unit/services/toast-test.ts:3]: Always writing tests for the moment ^^

**Review by Miexil — APPROVED** [review]

File: .storybook/preview-head.html (hunk 1 above)
Comment (JulienVannier66) [reply .storybook/preview-head.html]: I remove the file and set in the `.gitignore`. It will automatically generate

**Review by phndiaye — APPROVED** [review]

**Pull request merged**